consensus/utils.go

```diff
@@ -20,19 +20,53 @@ func IsValidSuperMajorityQuorumCertificate(qc QuorumCertificate, validators []Va
 	// Compute the signature that validators in the QC would have signed
 	signaturePayload := GetVoteSignaturePayload(qc.GetView(), qc.GetBlockHash())
 
+	isValidSuperMajority, validatorPublicKeysInQC := checkSuperMajorityStakeForAggregatedSignature(qc.GetAggregatedSignature(), validators)
+	if !isValidSuperMajority {
+		return false
+	}
+
+	return isValidSignatureManyPublicKeys(validatorPublicKeysInQC, qc.GetAggregatedSignature().GetSignature(), signaturePayload[:])
+}
+
+func IsValidSuperMajorityAggregateQuorumCertificate(aggQC AggregateQuorumCertificate, validators []Validator) bool {
+	if !isProperlyFormedAggregateQC(aggQC) || !isProperlyFormedValidatorSet(validators) {
+		return false
+	}
+
+	isValidSuperMajority, signerPublicKeys := checkSuperMajorityStakeForAggregatedSignature(aggQC.GetAggregatedSignature(), validators)
+	if !isValidSuperMajority {
+		return false
+	}
+
+	signedPayloads := [][]byte{}
```

Comment:
Can you add a comment here on why we do the payload aggregation this way? There's a nice comment in pos_network.go that you can refer to on how the highQC views are stored and how the payloads are signed: https://github.com/deso-protocol/core/blob/feature/proof-of-stake/lib/pos_network.go#L481